integration-tests/deploy_R_models.py

```diff
@@ -0,0 +1,192 @@
+            query_string, parsed_output))
+    return default
+
+
+def predict_R_model(df):
+    """
+    This function splits a dataframe into subframes consisting of a
+    single row. Each subframe is then csv-encoded and sent to Clipper
+    as a prediction request.
+    """
+    num_preds = len(df)
+    num_defaults = 0
+    for i in range(0, num_preds):
+        subframe = df.iloc[i:i + 1]
+        query_string = subframe.to_csv(sep=";")
+        num_defaults += call_predictions(query_string)
+    return num_preds, num_defaults
+
+
+def deploy_and_test_model(clipper, model, version, test_data_collection):
+    """
+    Parameters
+    ----------
+    test_data_collection : dict
+        A collection of pandas dataframes for which to request predictions
+    """
+    clipper.deploy_R_model(model_name, version, model,
+                           "clipper/r_python_container:latest", "string")
+    time.sleep(25)
+    num_preds = 0
+    num_defaults = 0
+    for i in range(0, len(test_data_collection)):
+        new_preds, new_defaults = predict_R_model(test_data_collection[i])
+        num_preds += new_preds
+        num_defaults += new_defaults
+
+    if num_defaults > 0:
+        print("Error: %d/%d predictions were default" % (num_defaults,
+                                                         num_preds))
+
+    print("PREDS: {} DEFAULTS: {}".format(num_preds, num_defaults))
+
+    if num_defaults > num_preds / 2:
+        raise BenchmarkException("Error querying APP %s, MODEL %s:%d" %
+                                 (app_name, model_name, version))
+
+
+def cleanup(clipper, test_succeeded):
+    """
+    Parameters
+    ----------
+    test_succeeded: bool
+        True if the test succeeded, False otherwise
+    """
+    clipper.stop_all()
+    if test_succeeded:
+        print("ALL TESTS PASSED")
+    else:
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    try:
+        clipper = init_clipper()
+
+        #preparing datasets for training and testing 
+        #using dataset mtcars , already provided by R. It has 32 rows and various coloums for eg. mpg,wt,cyl etc  
+        #splitting it for training and testing in ratio 1:1. Further splitting the test data in ratio 1:1
+        train_data = ro.r('train_data=head(mtcars,0.5*nrow(mtcars))')
+        test_data = ro.r('test_data=tail(mtcars,0.5*nrow(mtcars))')
+        test_data_collection = {}
+        test_data_collection[0] = ro.r(
+            'test_data1=head(test_data,0.5*nrow(test_data))')
+        test_data_collection[1] = ro.r(
+            'test_data2=tail(test_data,0.5*nrow(test_data))')
+
+        try:
+            clipper.register_application(app_name, model_name, "string",
+                                         "default_pred", 100000)
+            time.sleep(1)
+            response = requests.post(
+                "http://localhost:1337/%s/predict" % app_name,
+                headers=headers,
+                data=json.dumps({
+                    'uid': 0,
```

Comment:
You're likely using an old Docker container. Try pulling the `clipper/query_frontend` docker container from docker hub again.